src/handler/mod.rs

```diff
@@ -0,0 +1,151 @@
+/// This module most of low-level commands
+/// and grub variable modifications
+use anyhow::{bail, Error, Result};
+use std::fs::OpenOptions;
+use std::io::Write;
+use std::process::Command;
+use std::str;
+
+/// reboots the system if boot_counter is greater 0 or can be forced too
+pub fn handle_reboot(force: bool) -> Result<(), Error> {
+    if !force {
+        match get_boot_counter() {
+            Some(t) if t <= 0 => bail!("countdown ended, check greenboot-rollback status"),
+            None => bail!("boot_counter is not set"),
+            _ => {}
+        }
+    }
+    log::info!("restarting system");
+    let status = Command::new("systemctl").arg("reboot").status()?;
+    if status.success() {
+        return Ok(());
+    }
+    bail!("systemd returned error");
+}
+
+/// rollback to previous ostree deployment if boot counter is les than 0
+pub fn handle_rollback() -> Result<(), Error> {
+    match get_boot_counter() {
+        Some(t) if t <= 0 => {
+            log::info!("Greenboot will now attempt rollback");
+            let status = Command::new("rpm-ostree").arg("rollback").status()?;
+            if status.success() {
+                return Ok(());
+            }
+            bail!(status.to_string());
+        }
+        _ => log::info!("Rollback not initiated as boot_counter is either unset or not equal to 0"),
+    }
+    Ok(())
+}
+
+/// sets grub variable boot_counter if not set
+pub fn set_boot_counter(reboot_count: i32) -> Result<()> {
+    match get_boot_counter() {
+        Some(counter) => {
+            log::info!("boot_counter={counter}");
+            Ok(())
+        }
+        None => {
+            if set_grub_var("boot_counter", reboot_count) {
+                log::info!("boot_counter={reboot_count}");
+                return Ok(());
+            }
+            bail!("grub returned error");
```

Comment:
`bail!("grub returned error");` > grub returned _an_ error 

There are a few instances of this below as well.